translator/translate/otel/exporter/awsemf/translator.go

```diff
@@ -142,14 +140,27 @@ func (t *translator) Translate(c *confmap.Conf) (component.Config, error) {
 
 func getAppSignalsConfig() string {
 	ctx := context.CurrentContext()
-	kubernetesMode := ctx.KubernetesMode()
 
-	if kubernetesMode == config.ModeEKS {
+	mode := ctx.KubernetesMode()
+	if mode == "" {
+		mode = ctx.Mode()
+	}
+	if mode == config.ModeEC2 {
+		if ecsutil.GetECSUtilSingleton().IsECS() {
```

Comment:
as mentioned offline, we can keep the ecs translation bits since we still want customers on ecs to have a generic config. But let's remove the ECS resolver in the processor and schema changes until those are finalized.

Reply:
Updated.